docker-compose.override.yml

```diff
@@ -0,0 +1,31 @@
+version: "3.4"
+services:
+  nginx:
+    ports:
+      - "3000:80"
+
+  webapp:
+    build:
+      context: ./webapp
+      target: development
+    volumes:
+      - ./webapp/src:/webapp/src:ro
```

Comment:
why mount it as a volume instead of copying the folder inside the container?
AFAIK, mounting it as a volume is kind of a shortcut when you need persistency, which links to statefulness. Do we need statefulness here?

Reply:
Thank you for asking this.

Docker has many types of volumes (volume, bind mounts, tmpfs mounts, etc.).

This one here is a [bind mount](https://docs.docker.com/storage/bind-mounts/), and we use it to mount the src directory from the host into the container.

Mounting the directory allows us to modify the code from the host while npm reloads the code on change inside the container without having to rebuild the image.

Reply:
I'm understood that earlier, but I'm challenging the concept:

Why do we need to reload the code on change?
It is this only meant as a development shortcut?

Generally you have a program running without necessarily a state, and then the database behind it to preserve the state. Once your program "ships into production", it's meant as an off the shelf product: you don't change the code anymore. You only change the data.

Reply:
Yes, this is a development shortcut :+1:. docker-compose.override.yml has the configuration for the development environment.